### PR TITLE
added command to generate nats-server configuration for mem-resolver.

### PR DESCRIPTION
**Pull request by aricart**

### CODE REVIEW

File: cmd/memresolverconfigbuilder.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+		}
+		buf.WriteString(fmt.Sprintf("operator: %s\n", ofp))
+	} else if ofp == "--" {
+		buf.WriteString("# operator: <specify_path_to_operator_jwt>\n")
+	}
+	buf.WriteString("resolver: MEMORY\n")
+	buf.WriteString("resolver_preload: {\n")
+	for _, k := range keys {
+		v := cb.claims[k]
+		buf.WriteString(fmt.Sprintf("  %s: %s\n", k, v))
```
Comment (derekcollison) [inline cmd/memresolverconfigbuilder.go]: Maybe put jwt in "".

File: cmd/memresolverconfigbuilder_test.go [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+	require.NoError(t, err)
+	err = builder.Add(a)
+	require.NoError(t, err)
+
+	d, err := builder.Generate("")
+	require.NoError(t, err)
+
+	conf := string(d)
+	require.Contains(t, conf, "resolver: MEMORY")
+	require.Contains(t, conf, "resolver_preload: {")
```
Comment (derekcollison) [inline cmd/memresolverconfigbuilder_test.go]: Maybe have the server code parse it correctly and look at the opts it generates?

File: cmd/nkeyconfigbuilder_test.go [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+	require.NoError(t, err)
+
+	d, err := builder.Generate("")
+	require.NoError(t, err)
+
+	conf := string(d)
+	conf = strings.ReplaceAll(conf, " ", "")
+	conf = strings.ReplaceAll(conf, "\n", "")
+
+	require.Contains(t, conf, "accounts:{")
```
Comment (derekcollison) [inline cmd/nkeyconfigbuilder_test.go]: Use opts.go from server IMO.

File: cmd/memresolverconfigbuilder_test.go (hunk 2 above)
Comment (aricart) [reply cmd/memresolverconfigbuilder_test.go]: yes added a test for this, in this case I am testing that my code is generating what I expect to be there.

File: cmd/nkeyconfigbuilder_test.go (hunk 3 above)
Comment (aricart) [reply cmd/nkeyconfigbuilder_test.go]: yeap - added test.

File: cmd/memresolverconfigbuilder.go (hunk 1 above)
Comment (aricart) [reply cmd/memresolverconfigbuilder.go]: fixed

File: cmd/memresolverconfigbuilder_test.go (hunk 2 above)
Comment (derekcollison) [reply cmd/memresolverconfigbuilder_test.go]: I mean actually have the server's parser parse the config for validity.

File: cmd/memresolverconfigbuilder_test.go [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+
+func Test_MemResolverOperatorPlaceholder(t *testing.T) {
+	builder := NewMemResolverConfigBuilder()
+	d, err := builder.Generate("--")
+	require.NoError(t, err)
+	conf := string(d)
+	require.Contains(t, conf, "# operator: <specify_path_to_operator_jwt>\n")
+}
+
+func Test_MemResolverServerParse(t *testing.T) {
```
Comment (aricart) [inline cmd/memresolverconfigbuilder_test.go]: This is the server parse test.

File: cmd/memresolverconfigbuilder_test.go (hunk 2 above)
Comment (aricart) [reply cmd/memresolverconfigbuilder_test.go]: yes - I added a test, see last comment.

File: cmd/memresolverconfigbuilder_test.go (hunk 4 above)
Comment (aricart) [reply cmd/memresolverconfigbuilder_test.go]:
```	_, _, err := ExecuteCmd(createServerConfigCmd(), "--mem-resolver",
		"--config-file", serverconf,
		"--operator-jwt", opjwt)

 	require.NoError(t, err)

 	var opts server.Options
	require.NoError(t, opts.ProcessConfigFile(serverconf))
```

Comment (derekcollison) [reply cmd/memresolverconfigbuilder_test.go]: Gotcha, sorry I missed that.

**Pull request merged**